java/code/src/com/suse/manager/webui/utils/gson/ProxyContainerConfigJson.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.utils.gson;
+
+import com.suse.manager.ssl.SSLCertData;
+import com.suse.manager.ssl.SSLCertPair;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents the data sent from the UI to request a new proxy container configuration
+ */
+public class ProxyContainerConfigJson {
+
+    private static final String CREATE_SSL = "create-ssl";
+    private static final String USE_SSL = "use-ssl";
+
+    @SerializedName("proxyFQDN")
+    private String proxyFqdn;
+    private Integer proxyPort;
+    @SerializedName("serverFQDN")
+    private String serverFqdn;
```

Comment:
This parameter I am not sending it from the webUI, just to prevent the user typos that leads to a broken configuration. We should be able to get the `server fqdn` from the inside right?

Reply:
yes, we should, let me investigate that. In the worse case it's in the request itself